mate-panel/panel-action-button.c

```diff
@@ -213,13 +218,202 @@ panel_action_logout (GtkWidget *widget)
 						      PANEL_SESSION_MANAGER_LOGOUT_MODE_NORMAL);
 }
 
+/* Shutdown
+ */
+#ifdef HAVE_WAYLAND
 static void
-panel_action_shutdown (GtkWidget *widget)
+show_error_dialog(GtkWidget *label)
 {
-	PanelSessionManager *manager;
+	GtkWidget *dialog, *box, *content_area, *button, *image;
+	dialog = gtk_dialog_new();
+	button = gtk_dialog_add_button (GTK_DIALOG(dialog),
+			       "Cancel",
+			       GTK_RESPONSE_CANCEL);
+
+	image = gtk_image_new_from_icon_name("process-stop", GTK_ICON_SIZE_MENU);
+	gtk_button_set_image (GTK_BUTTON (button), image);
+
+	content_area = gtk_dialog_get_content_area (GTK_DIALOG (dialog));
+	box = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 6);
+	gtk_box_pack_start (GTK_BOX (box), label, TRUE, TRUE, 6);
+	gtk_container_set_border_width (GTK_CONTAINER (box), 16);
+	gtk_container_add (GTK_CONTAINER (content_area), box);
+	g_signal_connect_swapped (dialog, "response",
+				  G_CALLBACK (gtk_widget_destroy),
+				  dialog);
+
+	gtk_widget_show_all (dialog);
+}
+static void wayland_shutdown_response_cb (GtkWidget *dialog, guint response_id)
+	{
+		int ret;
 
-	manager = panel_session_manager_get ();
-	panel_session_manager_request_shutdown (manager);
+		if (response_id == GTK_RESPONSE_CANCEL)
+		{
+			gtk_widget_destroy(dialog);
+			return;
+		}
+
+		if (response_id == GTK_RESPONSE_OK)
+		{
+			gtk_widget_destroy(dialog);
```

Comment:
same here
```suggestion
			gtk_widget_destroy (dialog);
```

Reply:
Found and fixed in the latest force-push